support-frontend/assets/components/thankYou/thankYouModuleData.tsx

```diff
@@ -96,14 +113,23 @@ export const getThankYouModuleData = (
 		},
 		feedback: {
 			icon: getThankYouModuleIcon('feedback'),
-			header: getFeedbackHeader(feedbackSurveyHasBeenCompleted),
+			header: getFeedbackHeader(feedbackSurveyCompleted),
 			bodyCopy: (
 				<FeedbackBodyCopy
-					feedbackSurveyHasBeenCompleted={feedbackSurveyHasBeenCompleted}
+					feedbackSurveyHasBeenCompleted={feedbackSurveyCompleted}
 				/>
 			),
-			ctas: feedbackSurveyHasBeenCompleted ? null : (
-				<FeedbackCTA feedbackSurveyLink={getFeedbackSurveyLink(countryId)} />
+			ctas: feedbackSurveyCompleted ? null : (
+				<FeedbackCTA
+					feedbackSurveyLink={getFeedbackSurveyLink(countryId)}
+					onClick={() => {
+						SetFeedbackSurveyCompleted(true);
+						if (feedbackSurveyHasBeenCompleted) {
+							const dispatch = useContributionsDispatch();
+							dispatch(setThankYouFeedbackSurveyHasBeenCompleted(true));
+						}
+					}}
+				/>
```

Comment:
Nice refactor to avoid having to introduce complexity to our new setup!

Reply:
Thankyou, was hoping you would approve putting it there :)